CMakeLists.txt

```diff
@@ -408,7 +366,12 @@ set(Boost_ADDITIONAL_VERSIONS
 # TODO for the brave: transition all mlpack's CMake to 'target-based modern
 # CMake'.  Good luck!  You'll need it.
 set(Boost_NO_BOOST_CMAKE 1)
-find_package(Boost "${BOOST_VERSION}")
+if(DISABLE_DOWNLOADS)
+  find_package(Boost "${BOOST_VERSION}" REQUIRED)
+elseif(NOT Boost_FOUND AND (NOT DISABLE_DOWNLOADS))
```

Comment:
I think you have to have the `find_package()` here---if `DISABLE_DOWNLOADS` is `OFF`, then before this line we have not called `find_package(Boost ...)` at all.